podaac/swodlr_common/utilities.py

```diff
@@ -3,16 +3,86 @@
 import json
 import logging
 from importlib import resources
-from os import getenv
+from os import getenv, path
+import re
 from tempfile import NamedTemporaryFile
+from urllib.parse import urlparse
 
 import boto3
 import fastjsonschema
+from elasticsearch import Elasticsearch
 from requests import Session
 
 import podaac.swodlr_common
 
 
+class _SemVer:
```

Comment:
Did you consider trying https://pypi.org/project/semver/?

Reply:
I did, but it's parsing is kinda strict and I wanted the parsing on this feature to be a little bit more forgiving about where the version was in the string.

https://github.com/python-semver/python-semver/blob/68d19f59660c1c9b29a11f1403317288a1b5ccab/src/semver/version.py#L85-L103